### PR TITLE
Port logging articles - adapt to .NET docs

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

- Add code samples that correspond to using the .NET logging
- Add Microsoft Extensions logging articles
- Update TOC entry to add logging overview, providers, implement custom provider, and logger message

Contributes to: #4567

## Preview

✔️ [Logging in .NET](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/logging?branch=pr-en-us-20782)
✔️ [Logging providers in .NET](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/logging-providers?branch=pr-en-us-20782)
✔️ [Implement a custom logging provider in .NET](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/custom-logging-provider?branch=pr-en-us-20782)
✔️ [High-performance logging in .NET](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/high-performance-logging?branch=pr-en-us-20782)

### CODE REVIEW

File: docs/core/extensions/logging.md [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+
+## Logging output from dotnet run and Visual Studio
+
+Logs created with the [default logging providers](logging-providers.md) are displayed:
+
+- In Visual Studio
+  - In the Debug output window when debugging.
+- In the console window when the app is run with `dotnet run`.
+
+Logs that begin with "Microsoft" categories are from .NET code, and application code use the same logging API and providers.
```
Comment (tdykstra) [inline docs/core/extensions/logging.md:139]: This sentence is garbled.

File: docs/core/extensions/logging.md [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+- If multiple rules are selected, take the **last** one.
+- If no rules are selected, use `MinimumLevel`.
+
+## Logging output from dotnet run and Visual Studio
+
+Logs created with the [default logging providers](logging-providers.md) are displayed:
+
+- In Visual Studio
+  - In the Debug output window when debugging.
+- In the console window when the app is run with `dotnet run`.
```
Comment (tdykstra) [inline docs/core/extensions/logging.md:137]: Here it's not clear what group "default providers" refers to -- those loaded by CreateDefaultBuilder or all of the built-in ones?  It's definitely not true of all the built-in ones, and I suspect it's only true of Debug and Console providers.

**Review by tdykstra — CHANGES_REQUESTED** [review]
This is more great content.  It covers a lot of ground, but unless I missed it somewhere, it omits non-host content. Without that, for non-web non-host apps, devs still have to go to the ASP.NET Core docs for help. See https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-3.1#non-host-console-app and https://docs.microsoft.com/en-us/aspnet/core/migration/logging-nonaspnetcore?view=aspnetcore-3.1

File: docs/core/extensions/logging-providers.md [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+On Linux, the `Debug` provider log location is distribution-dependent and may be one of the following:
+
+- */var/log/message*
+- */var/log/syslog*
+
+### Event Source
+
+The `EventSource` provider writes to a cross-platform event source with the name `Microsoft-Extensions-Logging`. On Windows, the provider uses [ETW](/windows/win32/etw/event-tracing-portal).
+
+#### dotnet trace tooling
```
[the remark this answers is not in the document]
Comment (IEvangelist) [reply docs/core/extensions/logging-providers.md:92]: > this is a gigantic section on dotnet trace, should it go in the dotnet trace documentation or is that documentation already adequate?

That's a good callout. I'll look into that and merge what makes sense, or cut it.

File: docs/core/extensions/logging.md [hunk 4]
```diff
@@ -0,0 +1,515 @@
...
+- The `Debug` logging provider.
+- Log level `Information` and higher.
+- All categories starting with `"Microsoft"`.
+
+## See also
+
+- [Logging providers in .NET](logging-providers.md).
+- [Implement a custom logging provider in .NET](custom-logging-provider.md).
+- [High-performance logging in .NET](high-performance-logging.md).
+- Logging bugs should be created in the [github.com/dotnet/runtime/](https://github.com/dotnet/runtime/issues) repo.
```
[the remark this answers is not in the document]
Comment (IEvangelist) [reply docs/core/extensions/logging.md:515]: I'll fix that, good catch.

**Review by tdykstra — COMMENTED** [review]
The updates look good, but I still don't see non-host content.

**Comment by IEvangelist** [comment]
> The updates look good, but I still don't see non-host content.

Ah, yes... I forgot that bit. I'll add it, thanks.

File: docs/core/extensions/logging.md [hunk 5]
```diff
@@ -0,0 +1,529 @@
...
+title: Logging in .NET
+author: IEvangelist
+description: Learn how to use the logging framework provided by the Microsoft.Extensions.Logging NuGet package.
+ms.author: dapine
+ms.date: 09/30/2020
+---
+
+# Logging in .NET
+
+.NET supports a logging API that works with a variety of built-in and third-party logging providers. This article shows how to use the logging API with built-in providers. The code examples shown in this article apply to any .NET app that uses the [Generic Host](generic-host.md).
```
Comment (tdykstra) [inline docs/core/extensions/logging.md:11]:
```suggestion
.NET supports a logging API that works with a variety of built-in and third-party logging providers. This article shows how to use the logging API with built-in providers. Most of the code examples shown in this article apply to any .NET app that uses the [Generic Host](generic-host.md). For apps that don't use the Generic Host, see [Non-host console app](#non-host-console-app).
```

**Review by tdykstra — APPROVED** [review]
One additional change and this is ready to go. 😄

**Comment by gewarren** [comment]
There were some warnings introduced here. Can you fix them @IEvangelist ?

**Comment by IEvangelist** [comment]
> There were some warnings introduced here. Can you fix them @IEvangelist David Pine FTE ?

Good catch, thank you - see https://github.com/dotnet/docs/pull/20875

**Pull request merged**